meshnets/process.py

```diff
@@ -0,0 +1,67 @@
+"""The main file for data processing.
+
+It processes mesh files from a given directory and produce graph data
+as '.pt' files in another directory.
+"""
+
+import os
+from pathlib import Path
+
+from absl import app
+from absl import flags
+from absl import logging
+import torch
+
+from meshnets.utils import data_processing
+
+# Wind vector associated with the simulations
+WIND_VECTOR = (10, 0, 0)
+
+FLAGS = flags.FLAGS
+
+# Path flags
+flags.DEFINE_string('mesh_data_dir', os.path.join('data', 'vtk'),
+                    'Path to the folder containing the mesh files.')
+flags.DEFINE_string('processed_data_dir', os.path.join('data', 'pt'),
+                    'Path to the folder for the processed data files.')
+
+flags.DEFINE_bool('get_pressure', True,
+                  'Wether or not to extract pressure from the mesh files.')
+
+flags.DEFINE_bool(
+    'verbose', False,
+    'Wether or not the data processing should log processing information.')
+
+
+def main(_):
+    """Loop through a directory containing mesh files, produce a
+    graph representation of each mesh and save the graph as
+    a `.pt` file in the processed data directory."""
+
+    mesh_data_dir = FLAGS.mesh_data_dir
+    processed_data_dir = FLAGS.processed_data_dir
+    get_pressure = FLAGS.get_pressure
+    verbose = FLAGS.verbose
+
+    if not os.path.exists(processed_data_dir):
+        os.makedirs(processed_data_dir)
+
+    for mesh_file in os.listdir(mesh_data_dir):
+
+        mesh_file_path = os.path.join(mesh_data_dir, mesh_file)
+        processed_graph = data_processing.mesh_file_to_graph_data(
+            mesh_file_path,
+            WIND_VECTOR,
+            get_pressure=get_pressure,
```

Comment:
In which circumstance would you not want to get the pressure? Should the logic behind the data loading functions allow selecting whether or not to load velocity **and** whether or not to load pressure?

Reply:
The logic with this is that I am providing the mesh with already the pressure in. Ideally, one can provide one input file with just the mesh, and another with the same mesh and pressure measurements. But as of now, I am just providing the latter on, and he can extract the input from this.

Reply:
This also allows to process data for pure inference, where no pressure has been computed and is thus not present in the file.

Reply:
Right, so the graph with spatial coordinates as node features is always loaded and pressure is only loaded when **get_pressure=True**?

Reply:
Exactly. Would you find this clearer if I change the argument to `load_pressure`?

Reply:
Got it. No need to rename the argument, I just wanted to understand what was going in this data loading function.